FAIL: TestE2E/WeightedRoute

```
--- FAIL: TestE2E/WeightedRoute (0.76s)
        --- PASS: TestE2E/WeightedRoute/SameWeight (0.22s)
        --- PASS: TestE2E/WeightedRoute/BlueGreen (0.15s)
        --- PASS: TestE2E/WeightedRoute/CompleteRollout (0.11s)
        --- FAIL: TestE2E/WeightedRoute/MixedValidAndInvalid (0.10s)
```

https://github.com/envoyproxy/gateway/actions/runs/19357855098/job/55391844616

cc @zhaohuabing

--- a/test/e2e/tests/weighted_backend.go
+++ b/test/e2e/tests/weighted_backend.go
@@ -168,7 +168,7 @@ func testMixedValidAndInvalid(t *testing.T, suite *suite.ConformanceTestSuite) {
 		}
 	}
 
-	if !AlmostEquals(successCount, 40, 3) { // The weight of valid backend is 80%, so the expected success count is 50*80%=40
+	if !AlmostEquals(successCount, 40, 5) { // The weight of valid backend is 80%, so the expected success count is 50*80%=40
 		t.Errorf("The actual success count is not within the expected range, success %d", successCount)
 	}
 }

